includes/class-bcf-google-fonts-compatibility.php

```diff
@@ -0,0 +1,707 @@
+					// Add the file URL.
+					$result[ $font_family ][] = rtrim( ltrim( $match[0], 'url(' ), ')' );
+				}
+
+				// Make sure we have unique items.
+				// We're using array_flip here instead of array_unique for improved performance.
+				$result[ $font_family ] = array_flip( array_flip( $result[ $font_family ] ) );
+			}
+			return $result;
+		}
+
+		/**
+		 * Save setting - Sanitizes form inputs.
+		 *
+		 * @param array $input_settings setting data.
+		 * @return array    The sanitized form inputs.
+		 *
+		 * @since x.x.x
+		 */
+		public static function sanitize_form_inputs( $input_settings = array() ) {
+			$new_settings = array();
+
+			if ( ! empty( $input_settings ) ) {
+				foreach ( $input_settings as $key => $value ) {
+
+					$new_key = sanitize_text_field( $key );
+
+					if ( is_array( $value ) ) {
+						$new_settings[ $new_key ] = self::sanitize_form_inputs( $value );
+					} else {
+						$new_settings[ $new_key ] = sanitize_text_field( $value );
+					}
+				}
+			}
+
+			return $new_settings;
+		}
+
+		/**
+		 * Delete all Spectra font files from the theme JSON.
+		 *
+		 * @return void
+		 * @since x.x.x
+		 */
+		public static function delete_all_theme_font_family() {
+			if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() ) {
+				// Construct updated theme.json.
+				$theme_json_raw = json_decode( file_get_contents( get_stylesheet_directory() . '/theme.json' ), true );
+
+				// Overwrite the previous fontFamilies with the new ones.
+				$font_families = $theme_json_raw['settings']['typography']['fontFamilies'];
+
+				if ( ! empty( $font_families ) && is_array( $font_families ) ) {
+					$font_families = array_values(
+						array_filter(
+							$font_families,
+							function( $value ) {
+								if ( ! empty( $value['isBcf'] ) ) {
+									return false;
+								}
+								return true;
+							}
+						)
+					);
+				}
+
+				$theme_json_raw['settings']['typography']['fontFamilies'] = $font_families;
+
+				$theme_json        = wp_json_encode( $theme_json_raw, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE );
+				$theme_json_string = preg_replace( '~(?:^|\G)\h{4}~m', "\t", $theme_json );
+
+				// Write the new theme.json to the theme folder.
+				file_put_contents( // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions
+					get_stylesheet_directory() . '/theme.json',
+					$theme_json_string
+				);
+			}
+		}
+
+		/**
+		 * Delete font files from the CSS.
+		 *
+		 * @return void
+		 * @param array $font Font Data.
+		 * @since x.x.x
+		 */
+		public static function delete_theme_font_family( $font ) {
+			if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() ) {
+				// Construct updated theme.json.
+				$theme_json_raw = json_decode( file_get_contents( get_stylesheet_directory() . '/theme.json' ), true );
+
+				// Overwrite the previous fontFamilies with the new ones.
+				$font_families = $theme_json_raw['settings']['typography']['fontFamilies'];
```

Comment:
Check if settings and typography are available with a if condition @imnavanath . Do this for all similar instances in this file

Reply:
DONE